### PR TITLE
Sportspage

### PR DESCRIPTION
**Pull request by harthikss9**
will fetch all the events into sportspage , filters like available slots and unavailable slots, if slots count comes to 0, it will show details in unavailable tab,

### CODE REVIEW

File: lib/utilities/firebase_service.dart [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  // Register a user for an event
+  Future<void> registerForEvent(String eventId, String userId) async {
+    try {
+      final eventDoc = _firestore.collection('events-collection').doc(eventId);
+      await eventDoc.update({
+        'registeredUsers': FieldValue.arrayUnion([userId]),
+        'slotsAvailable': FieldValue.increment(-1), // Decrement available slots
+      });
+      print('User $userId registered for event $eventId');
```
Comment (jossean-rivera) [inline lib/utilities/firebase_service.dart:49]: Remove print statements

File: lib/utilities/firebase_service.dart [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    } catch (e) {
+      print('Error registering for event: $e');
```
Comment (jossean-rivera) [inline lib/utilities/firebase_service.dart:51]: Display an error to the user so they can try again later, you're just doing nothing.

File: lib/utilities/firebase_service.dart [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+class FirebaseService {
+  final FirebaseFirestore _firestore = FirebaseFirestore.instance;
```
Comment (jossean-rivera) [inline lib/utilities/firebase_service.dart:4]: Keep in mind the dependency injection pattern. You should had inject this service.

File: ios/Runner.xcodeproj/project.pbxproj
Comment (jossean-rivera) [inline ios/Runner.xcodeproj/project.pbxproj:1]: What are you changing this file for? Unless you're adding new packages, you can break the app.

File: lib/main.dart [hunk 4]
```diff
@@ -1,10 +1,8 @@
 import 'package:cloud_firestore/cloud_firestore.dart';
-import 'package:firebase_installations/firebase_installations.dart';
 import 'package:flutter/material.dart';
 import 'package:firebase_core/firebase_core.dart';
 import 'package:firebase_auth/firebase_auth.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
-import 'package:firebase_in_app_messaging/firebase_in_app_messaging.dart';
```
Comment (jossean-rivera) [inline lib/main.dart:7]: I don't think notifications are going to show up if you remove these. It was in the instructions to add the import for in-app messaging.

File: lib/navigation/router.dart [hunk 5]
```diff
@@ -53,10 +58,12 @@ GoRouter createRouter(AuthenticationBloc authenticationBloc) {
             builder: (context, state) => const SportsPage(),
             routes: [
               GoRoute(
-                path: RoutePaths.sportDetails,
+                path: ':sportId',
```
Comment (jossean-rivera) [inline lib/navigation/router.dart:61]: Add the constant to the RoutePaths, do not hard-code anything.

File: lib/pages/sport_details_page.dart [hunk 6]
```diff
@@ -1,25 +1,121 @@
...
-class SportDetailsPage extends StatelessWidget {
-  const SportDetailsPage({super.key});
+  const SportDetailsPage({super.key, required this.sportId});
+
+  @override
+  _SportDetailsPageState createState() => _SportDetailsPageState();
+}
+
+class _SportDetailsPageState extends State<SportDetailsPage> {
+  final FirebaseService firebaseService = FirebaseService();
```
Comment (jossean-rivera) [inline lib/pages/sport_details_page.dart:15]: DI pattern, inject an abstraction of the services.

File: lib/pages/sport_details_page.dart [hunk 7]
```diff
@@ -1,25 +1,121 @@
...
+        ],
       ),
-      body: Center(
-        child: FilledButton(
-          child: const Text("Go to back"),
-          onPressed: () {
-            context.goNamed(RouteNames.sports);
-          },
-        ),
+      body: FutureBuilder<List<Map<String, dynamic>>>(
```
Comment (jossean-rivera) [inline lib/pages/sport_details_page.dart:43]: You should return a some sport event object, not a map.

File: ios/Runner.xcodeproj/project.pbxproj
Comment (vamsi633) [reply ios/Runner.xcodeproj/project.pbxproj:1]: we are not changing anything , when we run this project with our pods , it is installing required ones . that's it

**Pull request merged**